src/shared/utilities/cliUtils.ts

```diff
@@ -234,88 +234,87 @@ function getToolkitLocalCliPath(): string {
 
 async function installSsmCli(
     tempDir: string,
-    progress: vscode.Progress<{ message?: string; increment?: number }>
+    progress: vscode.Progress<{ message?: string; increment?: number }>,
+    timeout: Timeout
 ): Promise<string> {
     progress.report({ message: msgDownloading })
-    const ssmInstaller = await downloadCliSource(AWS_CLIS['session-manager-plugin']!, tempDir)
+    const ssmInstaller = await downloadCliSource(AWS_CLIS['session-manager-plugin']!, tempDir, timeout)
     const outDir = path.join(getToolkitLocalCliPath(), 'sessionmanagerplugin')
     const finalPath = path.join(getToolkitLocalCliPath(), getOsCommand(AWS_CLIS['session-manager-plugin']!))
+    const TimedProcess = ChildProcess.extend({ timeout, rejectOnError: true })
 
     getLogger('channel').info(`Installing SSM CLI from ${ssmInstaller} to ${outDir}...`)
     progress.report({ message: msgInstallingLocal })
     switch (process.platform) {
         case 'darwin': {
-            return new Promise<string>(async (resolve, reject) => {
-                try {
-                    const tempPath = path.join(tempDir, 'tmp')
-                    const pkgArgs = ['--expand', 'session-manager-plugin.pkg', tempPath]
-                    const tarArgs = ['-xzf', path.join(tempPath, 'Payload')]
-                    await new ChildProcess('pkgutil', pkgArgs, { spawnOptions: { cwd: tempDir } }).run()
-                    await new ChildProcess('tar', tarArgs, { spawnOptions: { cwd: tempPath } }).run()
-
-                    fs.copySync(path.join(tempPath, 'usr', 'local', 'sessionmanagerplugin'), outDir, {
-                        recursive: true,
-                    })
-
-                    resolve(finalPath)
-                } catch (err) {
-                    reject(new InstallerError((err as Error).message))
+            try {
+                const tempPath = path.join(tempDir, 'tmp')
+                const pkgArgs = ['--expand', 'session-manager-plugin.pkg', tempPath]
+                const tarArgs = ['-xzf', path.join(tempPath, 'Payload')]
+                await new TimedProcess('pkgutil', pkgArgs).run({ spawnOptions: { cwd: tempDir } })
+                await new TimedProcess('tar', tarArgs).run({ spawnOptions: { cwd: tempPath } })
+
+                fs.copySync(path.join(tempPath, 'usr', 'local', 'sessionmanagerplugin'), outDir, {
+                    recursive: true,
+                })
+
+                return finalPath
+            } catch (err) {
+                if (TimeoutError.isCancelled(err)) {
+                    throw err
                 }
-            })
+                throw new InstallerError((err as Error).message)
```

Comment:
This block is confusing to me: if this is a cancellation (not a timeout), throw a `TimeoutError`, but if it is a timeout (not a cancellation) that has thrown a `TimeoutError`, throw an `InstallerError`?

Reply:
It's because user cancellation is not an error, but the process timing out is definitely an error.
Both 'events' use the same mechanism for control with different semantics. Renaming the error to `CancellationError` should clarify things I think. Then you exception handling checks the reason for the cancellation.